workflow/controller/operator.go

```diff
@@ -1934,6 +1934,27 @@ func (woc *wfOperationCtx) executeTemplate(ctx context.Context, nodeName string,
 		woc.updated = true
 	}
 
+	vols := make([]apiv1.PersistentVolumeClaim, 0)
+	for _, vol := range resolvedTmpl.VolumeClaimTemplates {
+		existed := false
+		for _, existedVol := range woc.execWf.Spec.VolumeClaimTemplates {
+			if existedVol.Name == vol.Name {
+				existed = true
+				break
+			}
+		}
+		if !existed {
+			vols = append(vols, vol)
+		}
+	}
+	if len(vols) > 0 {
+		woc.execWf.Spec.VolumeClaimTemplates = append(woc.execWf.Spec.VolumeClaimTemplates, vols...)
```

Comment:
execWf.Spec should be considered immutable (constant), this isn't good practice.

Reply:
okay, I might be able to put this part of the logic inside `setExecWorkflow(ctx context.Context)`, and reserve all the PVCs when initializing execWf.

Reply:
@Joibel https://github.com/argoproj/argo-workflows/blob/581950f7ae9f039a10d392d40a111f8f33e06eb3/workflow/controller/operator.go#L148-L150

`execWf` is a copy, modifying `execWf` is not good practice, but is this a feasible solution?